pkg/distributors/test/DistributionScheduler.test.ts

```diff
@@ -0,0 +1,209 @@
+
+    it('creates a scheduled distribution', async () => {
+      const time = (await currentTimestamp()).add(3600 * 24);
+      await scheduler
+        .connect(distributionOwner)
+        .scheduleDistribution(ZERO_BYTES32, stakingToken.address, distributionToken.address, amount, time);
+
+      const scheduleId = calcScheduleId(stakingToken, distributionToken, distributionOwner, time);
+
+      const response = await scheduler.getScheduledDistributionInfo(scheduleId);
+
+      expect(response.stakingToken).to.equal(stakingToken.address);
+      expect(response.distributionToken).to.equal(distributionToken.address);
+      expect(response.startTime).to.equal(time);
+      expect(response.owner).to.equal(distributionOwner.address);
+      expect(response.amount).to.equal(amount);
+      expect(response.status).to.equal(1);
+    });
+
+    it('transfers distributionTokens to the scheduler', async () => {
+      const time = (await currentTimestamp()).add(3600 * 24);
+
+      await expectBalanceChange(
+        () =>
+          scheduler
+            .connect(distributionOwner)
+            .scheduleDistribution(ZERO_BYTES32, stakingToken.address, distributionToken.address, amount, time),
+        distributionTokens,
+        [
+          { account: scheduler.address, changes: { [distributionToken.symbol]: amount } },
+          { account: distributionOwner.address, changes: { [distributionToken.symbol]: amount.mul(-1) } },
+        ]
+      );
+    });
+
+    it('emits a DistributionScheduled event', async () => {
+      const time = (await currentTimestamp()).add(3600 * 24);
+
+      const receipt = await (
+        await scheduler
+          .connect(distributionOwner)
+          .scheduleDistribution(ZERO_BYTES32, stakingToken.address, distributionToken.address, amount, time)
+      ).wait();
+
+      const scheduleId = calcScheduleId(stakingToken, distributionToken, distributionOwner, time);
+
+      expectEvent.inReceipt(receipt, 'DistributionScheduled', {
+        scheduleId,
+        owner: distributionOwner.address,
+        distributionToken: distributionToken.address,
+        startTime: time,
+        amount: amount,
+      });
+    });
+  });
+
+  describe('startDistributions', () => {
+    const amount = fp(1);
+
+    context('when distribution is pending', () => {
+      let scheduleId: string;
+      let time: BigNumber;
+      sharedBeforeEach(async () => {
+        // reward duration is important.  These tests assume a very short duration
+        time = (await currentTimestamp()).add(3600 * 24);
+        await scheduler
+          .connect(distributionOwner)
+          .scheduleDistribution(ZERO_BYTES32, stakingToken.address, distributionToken.address, amount, time);
+
+        scheduleId = calcScheduleId(stakingToken, distributionToken, distributionOwner, time);
+      });
+
+      // Skip tests pending support for paying into a distribution owned by another address
+      context.skip('when start time has passed', () => {
```

Comment:
We could also add a .skip test for when the start time has not passed

Reply:
Will handle later as mentioned above